File: Go/SelectionSort.go
//Go 实现选择排序

func selectionSort(arr []int)[]int {
    length := len(arr)
    for i := 0; i < length - 1; i++ {
        min := i
        for j := i + 1; j < length; j++ {
            if arr[j] < arr[min] {
                min = j
            }
        }
        arr[i],arr[min] = arr[min],arr[i]
    }
    return arr
}

File: Python/SelectionSort.py
# -*- coding = utf-8 -*-
# Python实现选择排序

def selectionSort(arr):
    for i in range(len(arr)-1):
        for j in range(i+1,len(arr)):
            if arr[j] < arr[i]:
                arr[i],arr[j] = arr[j],arr[i]
    return arr